caddyhttp/httpserver/server.go

```diff
@@ -413,7 +413,12 @@ func (s *Server) serveHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 	// the URL path, so a request to example.com/foo/blog on the site
 	// defined as example.com/foo appears as /blog instead of /foo/blog.
 	if pathPrefix != "/" {
-		r.URL.Path = strings.TrimPrefix(r.URL.Path, pathPrefix)
+		// We need to use URL.EscapedPath() when trimming the pathPrefix as
+		// URL.Path is ambiguous about / or %2f - see docs. See #1927
+		r.URL, err = r.URL.Parse(strings.TrimPrefix(r.URL.EscapedPath(), pathPrefix))
+		if err != nil {
```

Comment:
hmmm.  Do I need to check for err here or can I ignore it.  The likelyhood of the URL not being able to be parsed at this point is very minimal. 

Should I remove this check and error?

Reply:
I think golint reports unchecked errors, you can leave it like this.

Reply:
Better safe than sorry :)